conda/core/python_dist.py

```diff
@@ -0,0 +1,1103 @@
+        elif fname.endswith('.pth'):
+            continue
+        else:
+            continue
+        site_packages_anchor_files.add(anchor_file)
+
+    return site_packages_anchor_files
+
+
+def get_dist_file_from_egg_link(egg_link_file, prefix_path):
+    """
+    Return the egg info file path following an egg link.
+
+    Return `None` if no egg-info is found or the path is no longer there.
+    """
+    egg_info_full_path = None
+
+    with open(join(prefix_path, win_path_ok(egg_link_file))) as fh:
+        # See: https://setuptools.readthedocs.io/en/latest/formats.html#egg-links
+        # "...Each egg-link file should contain a single file or directory name
+        # with no newlines..."
+        egg_link_contents = fh.readlines()[0].strip()
+
+    egg_info_fnames = glob(join(egg_link_contents, "*.egg-info"))
+
+    if egg_info_fnames:
+        assert len(egg_info_fnames) == 1, (egg_link_file, egg_info_fnames)
+        egg_info_full_path = join(egg_link_contents, egg_info_fnames[0])
+
+        if isdir(egg_info_full_path):
+            egg_info_full_path = join(egg_info_full_path, "PKG-INFO")
+
+    return egg_info_full_path
+
+
+def get_python_distribution_info(anchor_file, prefix_path):
+    """
+    For a given anchor file return the python distribution.
+
+    Return `None` if the information was not found (can happen with egg-links).
+    """
+    if anchor_file.endswith('.egg-link'):
+        sp_reference = None
+        # This can be None in case the egg-info is no longer there
+        dist_file = get_dist_file_from_egg_link(anchor_file, prefix_path)
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_LINK
+    elif ".dist-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonInstalledDistribution
+        package_type = PackageType.SHADOW_PYTHON_DIST_INFO
+    elif anchor_file.endswith(".egg-info"):
+        sp_reference = basename(anchor_file)
+        dist_file = join(prefix_path, win_path_ok(anchor_file))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_FILE
+    elif ".egg-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_DIR
+    else:
+        raise NotImplementedError()
+
+    pydist = None
+
+    # An egg-link might reference a folder where egg-info is not available
+    if dist_file is not None:
+        try:
+            pydist = dist_cls(dist_file)
+        except Exception as error:
+            print('get_python_distribution_info', error)
+
+    return pydist, sp_reference, package_type
+
+
+def get_python_record(anchor_file, prefix_path, context):
```

Comment:
Move this function to the top, right after `get_python_records()` (see next comment)